src/darwin/Framework/CHIPTests/MTRDeviceTests.m

```diff
@@ -6123,6 +6137,58 @@ - (void)test046_MTRCommandWithRequiredResponseEncoding
     [self doEncodeDecodeRoundTrip:onCommand];
 }
 
+- (void)test047_DeviceMaybeReachableSetsUnreachable
+{
+    __auto_type * device = [MTRDevice deviceWithNodeID:@(kDeviceId2) controller:sController];
+    dispatch_queue_t queue = dispatch_get_main_queue();
+
+    MTRDeviceController * controller = sController;
+    XCTAssertNotNil(controller);
+
+    __auto_type * delegate = [[MTRDeviceTestDelegate alloc] init];
+
+    XCTestExpectation * subscribingExpectation = [self expectationWithDescription:@"Subscription attempt started"];
+
+    delegate.onInternalStateChanged = ^{
+        if ([device _getInternalState] == MTRInternalDeviceStateSubscribing) {
+            [subscribingExpectation fulfill];
+        }
+    };
+
+    [device setDelegate:delegate queue:queue];
+
+    // Start first subscription.
+    [device readAttributeWithEndpointID:@(1) clusterID:@(MTRClusterIDTypeColorControlID) attributeID:@(0) params:nil];
+
+    [self waitForExpectations:@[ subscribingExpectation ] timeout:60];
+
+    delegate.onInternalStateChanged = nil;
+
+    XCTestExpectation * notReachableExpectation = [self expectationWithDescription:@"Device is not reachable"];
+    delegate.onNotReachable = ^{
+        [notReachableExpectation fulfill];
+    };
+
+    // _deviceMayBeReachable should switch the device's state to notReachable.
```

Comment:
Please fix the PR description and commit message to match the actual API name?  The comments/code here have it right.